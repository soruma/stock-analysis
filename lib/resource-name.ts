export class ResourceName {
  public readonly systemName: string;
  public readonly systemEnv: string;

  constructor(systemName: string, systemEnv: string) {
    this.systemName = kebabToCamelCase(systemName, { capitalizeFirst: true });
    this.systemEnv = kebabToCamelCase(systemEnv, { capitalizeFirst: true });
  }

  private baseName(name: string): string {
    const str = kebabToCamelCase(name, { capitalizeFirst: true });
    return `${this.systemName}${str}${this.systemEnv}`;
  }

  public s3Name(name: string) {
    return `${camelToKebabCase(this.baseName(name))}-01941fc5-033f-7488-9a8c-8921a4298acc`;
  }

  public bucketPolicyName(name: string) {
    return this.baseName(name);
  }

  public kmsName(name: string) {
    return this.baseName(name);
  }

  public lambdaLayerVersionName(name: string) {
    return this.baseName(name);
  }

  public lambdaName(name: string) {
    return this.baseName(name);
  }

  public eventRoleName(name: string) {
    return this.baseName(name);
  }

  public stackName(name: string) {
    return this.baseName(name);
  }
}

export const camelToKebabCase = (str: string): string => {
  return str
    .replace(/([a-z])([A-Z])/g, '$1-$2')
    .replace(/([A-Z])([A-Z][a-z])/g, '$1-$2')
    .toLowerCase();
};

export const kebabToCamelCase = (input: string, options?: { capitalizeFirst?: boolean }): string => {
  const result = input
    .replace(/^-+|-+$/g, '')
    .split('-')
    .map((word, index) => {
      if (index === 0) {
        return word.toLowerCase();
      }
      return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
    })
    .join('');

  if (options?.capitalizeFirst) {
    return result.charAt(0).toUpperCase() + result.slice(1);
  }

  return result;
};
