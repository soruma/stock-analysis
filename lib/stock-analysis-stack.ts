import * as cdk from 'aws-cdk-lib';
import type { Construct } from 'constructs';
import { Runtime, Architecture } from 'aws-cdk-lib/aws-lambda';
import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
import { Bucket } from 'aws-cdk-lib/aws-s3';

import * as path from 'node:path';
import * as dotenv from 'dotenv';

import type { ResourceName } from './resource-name';
import { RetentionDays } from 'aws-cdk-lib/aws-logs';
import { Rule, RuleTargetInput, Schedule } from 'aws-cdk-lib/aws-events';
import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';

dotenv.config({ path: path.resolve(__dirname, '../.jquants.env') });

interface StockAnalysisLambdaStackProps extends cdk.StackProps {
    resourceName: ResourceName;
}

export class StockAnalysisStack extends cdk.Stack {
    constructor(scope: Construct, id: string, props: StockAnalysisLambdaStackProps) {
        super(scope, id, props);

        /***
         * S3
         */
        const s3 = new Bucket(this, props.resourceName.s3Name(), {
            bucketName: props.resourceName.s3Name(),
        });

        /***
         * Lambda
         */
        const downloadListedInfoFunction = new NodejsFunction(
            this,
            props.resourceName.lambdaName('download-listed-info'),
            {
                entry: 'assets/lambdas/download-listed-info/index.ts',
                handler: 'handler',
                architecture: Architecture.ARM_64,
                runtime: Runtime.NODEJS_22_X,
                timeout: cdk.Duration.seconds(30),
                environment: {
                    JQUANTS_API_MAIL_ADDRESS: process.env.JQUANTS_API_MAIL_ADDRESS || '',
                    JQUANTS_API_PASSWORD: process.env.JQUANTS_API_PASSWORD || '',
                    S3_BUCKET_NAME: s3.bucketName,
                },
                logRetention: RetentionDays.THIRTEEN_MONTHS,
            },
        );

        s3.grantPut(downloadListedInfoFunction);

        /***
         * EventBridge
         */
        const rule = new Rule(this, 'DailyRule', {
            schedule: Schedule.cron({ minute: '0', hour: '15' }),
        });

        rule.addTarget(
            new LambdaFunction(downloadListedInfoFunction, {
                event: RuleTargetInput.fromObject({
                    message: 'Scheduled Event',
                }),
            }),
        );
    }
}
