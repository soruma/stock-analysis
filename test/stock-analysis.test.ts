import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { expect, it } from 'vitest';

import { ResourceName } from '../lib/resource-name';
import { StockAnalysisStack } from '../lib/stock-analysis-stack';

const systemEnv = 'test';
const resourceName = new ResourceName('stock-snalysis', systemEnv);

it('snapshot test', () => {
  const app = new cdk.App();
  const stackName = resourceName.stackName('base');
  const stack = new StockAnalysisStack(app, stackName, {
    resourceName,
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
