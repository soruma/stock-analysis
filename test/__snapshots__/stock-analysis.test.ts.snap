// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
{
  Mappings: {
    LatestNodeRuntimeMap: {
      af-south-1: {
        value: nodejs20.x,
      },
      ap-east-1: {
        value: nodejs20.x,
      },
      ap-northeast-1: {
        value: nodejs20.x,
      },
      ap-northeast-2: {
        value: nodejs20.x,
      },
      ap-northeast-3: {
        value: nodejs20.x,
      },
      ap-south-1: {
        value: nodejs20.x,
      },
      ap-south-2: {
        value: nodejs20.x,
      },
      ap-southeast-1: {
        value: nodejs20.x,
      },
      ap-southeast-2: {
        value: nodejs20.x,
      },
      ap-southeast-3: {
        value: nodejs20.x,
      },
      ap-southeast-4: {
        value: nodejs20.x,
      },
      ap-southeast-5: {
        value: nodejs20.x,
      },
      ap-southeast-7: {
        value: nodejs20.x,
      },
      ca-central-1: {
        value: nodejs20.x,
      },
      ca-west-1: {
        value: nodejs20.x,
      },
      cn-north-1: {
        value: nodejs18.x,
      },
      cn-northwest-1: {
        value: nodejs18.x,
      },
      eu-central-1: {
        value: nodejs20.x,
      },
      eu-central-2: {
        value: nodejs20.x,
      },
      eu-isoe-west-1: {
        value: nodejs18.x,
      },
      eu-north-1: {
        value: nodejs20.x,
      },
      eu-south-1: {
        value: nodejs20.x,
      },
      eu-south-2: {
        value: nodejs20.x,
      },
      eu-west-1: {
        value: nodejs20.x,
      },
      eu-west-2: {
        value: nodejs20.x,
      },
      eu-west-3: {
        value: nodejs20.x,
      },
      il-central-1: {
        value: nodejs20.x,
      },
      me-central-1: {
        value: nodejs20.x,
      },
      me-south-1: {
        value: nodejs20.x,
      },
      mx-central-1: {
        value: nodejs20.x,
      },
      sa-east-1: {
        value: nodejs20.x,
      },
      us-east-1: {
        value: nodejs20.x,
      },
      us-east-2: {
        value: nodejs20.x,
      },
      us-gov-east-1: {
        value: nodejs18.x,
      },
      us-gov-west-1: {
        value: nodejs18.x,
      },
      us-iso-east-1: {
        value: nodejs18.x,
      },
      us-iso-west-1: {
        value: nodejs18.x,
      },
      us-isob-east-1: {
        value: nodejs18.x,
      },
      us-west-1: {
        value: nodejs20.x,
      },
      us-west-2: {
        value: nodejs20.x,
      },
    },
  },
  Parameters: {
    BootstrapVersion: {
      Default: /cdk-bootstrap/hnb659fds/version,
      Description: Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip],
      Type: AWS::SSM::Parameter::Value<String>,
    },
  },
  Resources: {
    DailyRule7A0FEF67: {
      Properties: {
        ScheduleExpression: cron(0 17 * * ? *),
        State: ENABLED,
        Targets: [
          {
            Arn: {
              Fn::GetAtt: [
                StockSnalysisDownloadListedInfoTestF33E65B2,
                Arn,
              ],
            },
            Id: Target0,
            Input: {
              Fn::Join: [
                ,
                [
                  {"message":"Scheduled event for ,
                  {
                    Ref: StockSnalysisDownloadListedInfoTestF33E65B2,
                  },
                  "},
                ],
              ],
            },
          },
          {
            Arn: {
              Fn::GetAtt: [
                StockSnalysisDownloadPricesDailyQuotesTest1F92F84A,
                Arn,
              ],
            },
            Id: Target1,
            Input: {
              Fn::Join: [
                ,
                [
                  {"message":"Scheduled event for ,
                  {
                    Ref: StockSnalysisDownloadPricesDailyQuotesTest1F92F84A,
                  },
                  "},
                ],
              ],
            },
          },
        ],
      },
      Type: AWS::Events::Rule,
    },
    DailyRuleAllowEventRuleStockSnalysisBaseTestStockSnalysisDownloadListedInfoTest067D3AC4A87C6FA8: {
      Properties: {
        Action: lambda:InvokeFunction,
        FunctionName: {
          Fn::GetAtt: [
            StockSnalysisDownloadListedInfoTestF33E65B2,
            Arn,
          ],
        },
        Principal: events.amazonaws.com,
        SourceArn: {
          Fn::GetAtt: [
            DailyRule7A0FEF67,
            Arn,
          ],
        },
      },
      Type: AWS::Lambda::Permission,
    },
    DailyRuleAllowEventRuleStockSnalysisBaseTestStockSnalysisDownloadPricesDailyQuotesTest934F69C98C4D3209: {
      Properties: {
        Action: lambda:InvokeFunction,
        FunctionName: {
          Fn::GetAtt: [
            StockSnalysisDownloadPricesDailyQuotesTest1F92F84A,
            Arn,
          ],
        },
        Principal: events.amazonaws.com,
        SourceArn: {
          Fn::GetAtt: [
            DailyRule7A0FEF67,
            Arn,
          ],
        },
      },
      Type: AWS::Lambda::Permission,
    },
    LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A: {
      DependsOn: [
        LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB,
        LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB,
      ],
      Properties: {
        Code: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: hashed.zip,
        },
        Handler: index.handler,
        Role: {
          Fn::GetAtt: [
            LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB,
            Arn,
          ],
        },
        Runtime: {
          Fn::FindInMap: [
            LatestNodeRuntimeMap,
            {
              Ref: AWS::Region,
            },
            value,
          ],
        },
        Timeout: 900,
      },
      Type: AWS::Lambda::Function,
    },
    LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB: {
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: sts:AssumeRole,
              Effect: Allow,
              Principal: {
                Service: lambda.amazonaws.com,
              },
            },
          ],
          Version: 2012-10-17,
        },
        ManagedPolicyArns: [
          {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole,
              ],
            ],
          },
        ],
      },
      Type: AWS::IAM::Role,
    },
    LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB: {
      Properties: {
        PolicyDocument: {
          Statement: [
            {
              Action: [
                logs:PutRetentionPolicy,
                logs:DeleteRetentionPolicy,
              ],
              Effect: Allow,
              Resource: *,
            },
          ],
          Version: 2012-10-17,
        },
        PolicyName: LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB,
        Roles: [
          {
            Ref: LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB,
          },
        ],
      },
      Type: AWS::IAM::Policy,
    },
    StockSnalysisDataBucketKeyTest447B7803: {
      DeletionPolicy: Retain,
      Properties: {
        EnableKeyRotation: true,
        KeyPolicy: {
          Statement: [
            {
              Action: kms:*,
              Effect: Allow,
              Principal: {
                AWS: {
                  Fn::Join: [
                    ,
                    [
                      arn:,
                      {
                        Ref: AWS::Partition,
                      },
                      :iam::,
                      {
                        Ref: AWS::AccountId,
                      },
                      :root,
                    ],
                  ],
                },
              },
              Resource: *,
            },
            {
              Action: [
                kms:Decrypt,
                kms:GenerateDataKey*,
              ],
              Effect: Allow,
              Principal: {
                AWS: arn:aws:iam::123456789012:role/example-role,
              },
              Resource: *,
              Sid: Grant Abacus.AI KMS access,
            },
          ],
          Version: 2012-10-17,
        },
      },
      Type: AWS::KMS::Key,
      UpdateReplacePolicy: Retain,
    },
    StockSnalysisDataBucketKeyTestAliasB8803B11: {
      Properties: {
        AliasName: alias/data-bucket-key,
        TargetKeyId: {
          Fn::GetAtt: [
            StockSnalysisDataBucketKeyTest447B7803,
            Arn,
          ],
        },
      },
      Type: AWS::KMS::Alias,
    },
    StockSnalysisDataBucketPolicyTest88A2A91C: {
      Properties: {
        Bucket: {
          Ref: stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
        },
        PolicyDocument: {
          Statement: [
            {
              Action: [
                s3:GetObject,
                s3:ListBucket,
              ],
              Effect: Allow,
              Principal: {
                AWS: arn:aws:iam::123456789012:role/example-role,
              },
              Resource: [
                {
                  Fn::Join: [
                    ,
                    [
                      {
                        Fn::GetAtt: [
                          stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
                          Arn,
                        ],
                      },
                      /*,
                    ],
                  ],
                },
                {
                  Fn::GetAtt: [
                    stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
                    Arn,
                  ],
                },
              ],
            },
          ],
          Version: 2012-10-17,
        },
      },
      Type: AWS::S3::BucketPolicy,
    },
    StockSnalysisDownloadListedInfoTestF33E65B2: {
      DependsOn: [
        StockSnalysisDownloadListedInfoTestServiceRoleDefaultPolicy730210DD,
        StockSnalysisDownloadListedInfoTestServiceRole90EA6BCB,
      ],
      Properties: {
        Architectures: [
          arm64,
        ],
        Code: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: hashed.zip,
        },
        Environment: {
          Variables: {
            JQUANTS_API_MAIL_ADDRESS: test@example.com,
            JQUANTS_API_PASSWORD: password,
            S3_BUCKET_NAME: {
              Ref: stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
            },
          },
        },
        Handler: index.handler,
        Layers: [
          {
            Ref: StockSnalysisStockAnalysisModulesTest94FBC596,
          },
          {
            Ref: StockSnalysisJQuantsTest4F63EB39,
          },
        ],
        Role: {
          Fn::GetAtt: [
            StockSnalysisDownloadListedInfoTestServiceRole90EA6BCB,
            Arn,
          ],
        },
        Runtime: nodejs22.x,
        Timeout: 30,
      },
      Type: AWS::Lambda::Function,
    },
    StockSnalysisDownloadListedInfoTestLogRetentionC1131700: {
      Properties: {
        LogGroupName: {
          Fn::Join: [
            ,
            [
              /aws/lambda/,
              {
                Ref: StockSnalysisDownloadListedInfoTestF33E65B2,
              },
            ],
          ],
        },
        RetentionInDays: 400,
        ServiceToken: {
          Fn::GetAtt: [
            LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A,
            Arn,
          ],
        },
      },
      Type: Custom::LogRetention,
    },
    StockSnalysisDownloadListedInfoTestServiceRole90EA6BCB: {
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: sts:AssumeRole,
              Effect: Allow,
              Principal: {
                Service: lambda.amazonaws.com,
              },
            },
          ],
          Version: 2012-10-17,
        },
        ManagedPolicyArns: [
          {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole,
              ],
            ],
          },
        ],
      },
      Type: AWS::IAM::Role,
    },
    StockSnalysisDownloadListedInfoTestServiceRoleDefaultPolicy730210DD: {
      Properties: {
        PolicyDocument: {
          Statement: [
            {
              Action: [
                s3:PutObject,
                s3:PutObjectLegalHold,
                s3:PutObjectRetention,
                s3:PutObjectTagging,
                s3:PutObjectVersionTagging,
                s3:Abort*,
              ],
              Effect: Allow,
              Resource: {
                Fn::Join: [
                  ,
                  [
                    {
                      Fn::GetAtt: [
                        stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
                        Arn,
                      ],
                    },
                    /*,
                  ],
                ],
              },
            },
            {
              Action: [
                kms:Encrypt,
                kms:ReEncrypt*,
                kms:GenerateDataKey*,
                kms:Decrypt,
              ],
              Effect: Allow,
              Resource: {
                Fn::GetAtt: [
                  StockSnalysisDataBucketKeyTest447B7803,
                  Arn,
                ],
              },
            },
          ],
          Version: 2012-10-17,
        },
        PolicyName: StockSnalysisDownloadListedInfoTestServiceRoleDefaultPolicy730210DD,
        Roles: [
          {
            Ref: StockSnalysisDownloadListedInfoTestServiceRole90EA6BCB,
          },
        ],
      },
      Type: AWS::IAM::Policy,
    },
    StockSnalysisDownloadPricesDailyQuotesTest1F92F84A: {
      DependsOn: [
        StockSnalysisDownloadPricesDailyQuotesTestServiceRoleDefaultPolicy6B92AE3A,
        StockSnalysisDownloadPricesDailyQuotesTestServiceRole05189745,
      ],
      Properties: {
        Architectures: [
          arm64,
        ],
        Code: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: hashed.zip,
        },
        Environment: {
          Variables: {
            JQUANTS_API_MAIL_ADDRESS: test@example.com,
            JQUANTS_API_PASSWORD: password,
            S3_BUCKET_NAME: {
              Ref: stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
            },
          },
        },
        Handler: index.handler,
        Layers: [
          {
            Ref: StockSnalysisStockAnalysisModulesTest94FBC596,
          },
          {
            Ref: StockSnalysisJQuantsTest4F63EB39,
          },
        ],
        Role: {
          Fn::GetAtt: [
            StockSnalysisDownloadPricesDailyQuotesTestServiceRole05189745,
            Arn,
          ],
        },
        Runtime: nodejs22.x,
        Timeout: 30,
      },
      Type: AWS::Lambda::Function,
    },
    StockSnalysisDownloadPricesDailyQuotesTestLogRetention9A8FFFEF: {
      Properties: {
        LogGroupName: {
          Fn::Join: [
            ,
            [
              /aws/lambda/,
              {
                Ref: StockSnalysisDownloadPricesDailyQuotesTest1F92F84A,
              },
            ],
          ],
        },
        RetentionInDays: 400,
        ServiceToken: {
          Fn::GetAtt: [
            LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A,
            Arn,
          ],
        },
      },
      Type: Custom::LogRetention,
    },
    StockSnalysisDownloadPricesDailyQuotesTestServiceRole05189745: {
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: sts:AssumeRole,
              Effect: Allow,
              Principal: {
                Service: lambda.amazonaws.com,
              },
            },
          ],
          Version: 2012-10-17,
        },
        ManagedPolicyArns: [
          {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole,
              ],
            ],
          },
        ],
      },
      Type: AWS::IAM::Role,
    },
    StockSnalysisDownloadPricesDailyQuotesTestServiceRoleDefaultPolicy6B92AE3A: {
      Properties: {
        PolicyDocument: {
          Statement: [
            {
              Action: [
                s3:PutObject,
                s3:PutObjectLegalHold,
                s3:PutObjectRetention,
                s3:PutObjectTagging,
                s3:PutObjectVersionTagging,
                s3:Abort*,
              ],
              Effect: Allow,
              Resource: {
                Fn::Join: [
                  ,
                  [
                    {
                      Fn::GetAtt: [
                        stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C,
                        Arn,
                      ],
                    },
                    /*,
                  ],
                ],
              },
            },
            {
              Action: [
                kms:Encrypt,
                kms:ReEncrypt*,
                kms:GenerateDataKey*,
                kms:Decrypt,
              ],
              Effect: Allow,
              Resource: {
                Fn::GetAtt: [
                  StockSnalysisDataBucketKeyTest447B7803,
                  Arn,
                ],
              },
            },
          ],
          Version: 2012-10-17,
        },
        PolicyName: StockSnalysisDownloadPricesDailyQuotesTestServiceRoleDefaultPolicy6B92AE3A,
        Roles: [
          {
            Ref: StockSnalysisDownloadPricesDailyQuotesTestServiceRole05189745,
          },
        ],
      },
      Type: AWS::IAM::Policy,
    },
    StockSnalysisJQuantsTest4F63EB39: {
      Properties: {
        CompatibleArchitectures: [
          arm64,
        ],
        Content: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: hashed.zip,
        },
      },
      Type: AWS::Lambda::LayerVersion,
    },
    StockSnalysisStockAnalysisModulesTest94FBC596: {
      Properties: {
        CompatibleArchitectures: [
          arm64,
        ],
        Content: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: hashed.zip,
        },
      },
      Type: AWS::Lambda::LayerVersion,
    },
    stocksnalysisdatatest01941fc5033f74889a8c8921a4298acc1EECB35C: {
      DeletionPolicy: Retain,
      Properties: {
        BucketEncryption: {
          ServerSideEncryptionConfiguration: [
            {
              ServerSideEncryptionByDefault: {
                KMSMasterKeyID: {
                  Fn::GetAtt: [
                    StockSnalysisDataBucketKeyTest447B7803,
                    Arn,
                  ],
                },
                SSEAlgorithm: aws:kms,
              },
            },
          ],
        },
        BucketName: stock-snalysis-data-test-01941fc5-033f-7488-9a8c-8921a4298acc,
      },
      Type: AWS::S3::Bucket,
      UpdateReplacePolicy: Retain,
    },
  },
  Rules: {
    CheckBootstrapVersion: {
      Assertions: [
        {
          Assert: {
            Fn::Not: [
              {
                Fn::Contains: [
                  [
                    1,
                    2,
                    3,
                    4,
                    5,
                  ],
                  {
                    Ref: BootstrapVersion,
                  },
                ],
              },
            ],
          },
          AssertDescription: CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.,
        },
      ],
    },
  },
}
`;
